otsd

```diff
@@ -111,6 +114,16 @@ elif args.btc_net == 'regtest':
     bitcoin.SelectParams('regtest')
 
 
+if args.btc_donation_address:
+    proxy = bitcoin.rpc.Proxy()
+    try:
+        # check validity of donation address
+        proxy.call('getaddressinfo', args.btc_donation_address)
```

Comment:
You can check validity with python-bitcoinlib; better than a RPC call. Since this is a command line option, an invalid Bitcoin address should fail immediately.

In fact, IIRC you can check validity above in `parser.add_argument()` so the type is a `CBitcoinAddress` rather than `str`

Reply:
It would be very elegant but bitcoin.SelectParams() is executed after parsing, so CBitcoinAddress will fail for testnet and regtest addresses.

Reply:
(I tried also to figure an action that use bitcoin.SelectParams() when --btc-???net params are parsed, but this implies a sorting problem...)